snapcraft/internal/build_providers/_qemu/_keys.py

```diff
@@ -0,0 +1,61 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+from typing import TypeVar, Type
+
+from paramiko import RSAKey
+
+from snapcraft.internal.errors import SnapcraftError
+
+
+SSHKeyT = TypeVar('SSHKeyT', bound='SSHKey')
+
+
+class SSHKeyPathError(SnapcraftError):
```

Comment:
This error class name could be more descriptive: it's not just a problem with the path (permissions etc.), it's that the path doesn't actually exist.

Reply:
Also, no strong feelings here, but I'm curious why this isn't in `build_providers/errors.py`?

Reply:
I had the feeling it was qemu specific, but after a check I see the telnet stuff made it there too.